sretoolbox/container/image.py

```diff
@@ -80,13 +81,19 @@ class Image:
     :param password: (optional) The private registry password
     :param auth_server: (optional) The host that the username and password are
                         meant for
+    :param response_cache: (optional) Provide a response cache that acts
+                           as a dict
     :param ssl_verify: (optional) Whether to verify the SSL certificate
     """
 
-    # 50KB is an arbitrary number, most manifests are on the order of
-    # 10-15KB, so 50KB gives us enough room for larger instances,
-    # while protecting us from OOM-ing too easily
-    MAX_CACHE_ITEM_SIZE = 50*1024
+    # This is a method dispatcher to handle how to handle responses that are
+    # already present in the response_cache
+    _HANDLE_RESPONSE_CACHE_METHODS = {
```

Comment:
This is begging for a separate class, but to keep it simple, you can just get rid of the `_handle_cache_response` by having the callables directly in the dict instead of strings:

```
_HANDLE_RESPONSE_CACHE_METHODS = {"docker.io": Image._handle_docker_content_digest, ...
```

Reply:
I also wanted to use callables instead of strings, but I wasn't able to make it work. Python doesn't like what you suggest...

```
sretoolbox/container/image.py:92: in Image
    "docker.io": Image._handle_docker_content_digest,
E   NameError: name 'Image' is not defined
```

I can look into a separate class, but I don't know how to get rid of the `getattr` thingy if I want to implement a dispatch table inside a class. I can remove the intermediate `_handle_cache_response` if that looks a bit redundant (it just uses the dispatch table, nothing else).

Reply:
Removed `_handle_cache_response` in 21859d8

Reply:
right, it's too early for the parser
well, we could move the _HANDLE_RESPONSE_CACHE_METHODS into the init...

```
    def __init__(...):
        self._cache_methods_map = {"docker.io": self._handle_docker_content_digest, ...}
```

anyway, let it be... the string identifiers make me nervous, but that's just me :)

Reply:
ok. I'll let it be as it is. I don't see it very much as an instance variable, TBH. Blame Guido for creating the `getattr` function :-P